packages/orn-locator/src/resolvers/books.ts

```diff
@@ -0,0 +1,155 @@
+import fetch from 'node-fetch';
+
+const oswebUrl = 'https://openstax.org/apps/cms/api/v2/pages';
+const fields = 'cnx_id,authors,publish_date,cover_color,amazon_link,book_state,promote_image,webview_rex_link,cover_url,title_image_url';
+
+const getArchiveInfo = async (bookId: string) => {
+  const releaseJson = await fetch('https://openstax.org/rex/release.json')
+    .then(response => response.json())
+  ;
+
+  const bookConfig = releaseJson.books[bookId];
+
+  return {
+    archivePath: bookConfig.archiveOverride || releaseJson.archiveUrl,
+    bookVersion: bookConfig.defaultVersion,
+  };
+};
+
+export const book = async(id: string) => {
+  const oswebData = await fetch(`${oswebUrl}?type=books.Book&fields=${fields}&cnx_id=${id}`)
+    .then(response => response.json() as any)
+    .then(data => data.items[0])
+  ;
+
+  return {
+    id,
+    orn: `https://openstax.org/orn/content:book/${id}`,
+    type: 'content:book',
+    title: oswebData.title,
+    slug: oswebData.meta.slug,
+    theme: oswebData.cover_color,
+    urls: {
+      default: oswebData.meta.html_url,
+      information: oswebData.meta.html_url,
+      experience: oswebData.webview_rex_link,
+    },
+    images: {
+      default: oswebData.cover_url,
```

Comment:
i tried to keep these fields sort of generic so that we can try to establish some patterns all resources might follow, but i'm not really formalizing it yet